src/proxyvisor.coffee

```diff
@@ -0,0 +1,208 @@
+
+router.get '/v1/devices/:uuid', (req, res) ->
+	uuid = req.params.uuid
+	knex('dependentDevice').select().where({ uuid })
+	.then ([ device ]) ->
+		return res.status(404).send('Device not found') if !device?
+		res.json(device)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.put '/v1/devices/:uuid/logs', (req, res) ->
+	uuid = req.params.uuid
+	m = {
+		message: req.body.message
+		timestamp: req.body.timestamp or Date.now()
+	}
+	m.isSystem = req.body.isSystem if req.body.isSystem?
+
+	knex('dependentDevice').select().where({ uuid })
+	.then ([ device ]) ->
+		return res.status(404).send('Device not found') if !device?
+		pubnub.publish({ channel: "device-#{device.logs_channel}-logs", message: m })
+	.then ->
+		res.status(202).send('OK')
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.put '/v1/devices/:uuid', (req, res) ->
+	uuid = req.params.uuid
+	status = req.body.status
+	is_online = req.body.is_online
+	utils.getConfig('apiKey')
+	.then (apiKey) ->
+		throw new Error('apikey not found') if !apiKey?
+		knex('dependentDevice').select().where({ uuid })
+		.then ([ device ]) ->
+			return res.status(404).send('Device not found') if !device?
+			resinApi.patch
+				resource: 'device'
+				id: device.deviceId
+				body: { status, is_online }
+				customOptions:
+					apikey: apiKey
+			.then ->
+				device.status = status
+				device.is_online = is_online
+				knex('dependentDevice').update(device).where({ uuid })
+				.then ->
+					res.json(device)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+tarPath = (app) ->
+	return '/tmp/' + app.commit + '.tar'
+
+router.get '/v1/assets/:commit', (req, res) ->
+	knex('dependentApp').select().where({ commit: req.params.commit })
+	.then ([ app ]) ->
+		return res.status(404).send('Not found') if !app
+		dest = tarPath(app)
+		getAssetsPath(app.imageId)
+		.then (path) ->
+			getTarArchive(path, dest)
+		.then ->
+			res.sendFile(dest)
+	.catch (err) ->
+		console.error(err)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+getTarArchive = (path, destination) ->
+	fs.lstatAsync(path)
+	.then ->
+		execAsync("cd #{path} && tar -cvf #{destination} *")
+
+# TODO: deduplicate code from compareForUpdate in application.coffee
+exports.fetchAndSetTargetsForDependentApps = (state, fetchFn) ->
+	knex('dependentApp').select()
+	.then (localDependentApps) ->
+		# Compare to see which to fetch, and which to delete
+		remoteApps = _.mapValues state.apps, (app, appId) ->
+			env = app.environment ? {}
+			return {
+				appId: appId
+				imageId: app.image
+				commit: app.commit
+				env: JSON.stringify(env)
+			}
+		localApps = _.indexBy(localDependentApps, 'appId')
+
+		toBeDownloaded = _.values(
+			_.filter remoteApps, (app, appId) ->
+				return  app.commit? and !_.any(localApps, imageId: app.imageId)
+		)
+		toBeRemoved = _.values(
```

Comment:
This `_.values` is unnecessary as `_.filter` already returns an array of values (if you wanted to keep it in object form you'd need to use `_.pickBy`, or `_.pick` for lodash v3)